### PR TITLE
Migrating bv-ui-pixels-displayed to module in the core

### PR DESCRIPTION
**Pull request by msmolev**
This is an attempt to migrate https://github.com/bazaarvoice/bv-ui-pixels-displayed into the core library.
Original library, however, only checked vertical visibility. So I have also duplicated the function and created horizontal version as well. 

Migrated tests, created horizontal tests, added link in Readme.

### CODE REVIEW

File: lib/pixelsDisplayed/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# pixelsDisplayed
+
+> A module for detecting whether an element is located within the viewport
+
+## Module arguments
+
+ - `window`: Reference to window.
```
Comment (reason-bv) [inline lib/pixelsDisplayed/README.md]: It won't need either of these, as it can use `global`.

File: lib/pixelsDisplayed/index.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+/**
+ * @fileOverview Checks visibility of an element by calculating the
+ * visible pixels for vertical and horizontal viewport.
+ * This is a migration of repository bv-ui-pixels-displayed
+ */
+module.exports = function (window, document) {
```
Comment (reason-bv) [inline lib/pixelsDisplayed/index.js]:
```
var global = import('../global');

module.exports = function () {
  var window = global;
  var document = window.document;
```

File: test/unit/pixelsDisplayed/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
+/**
```
Comment (reason-bv) [inline test/unit/pixelsDisplayed/index.spec.js]: Some small changes needed here to mock the the necessary items given use of global rather than arguments.

**Comment by msmolev** [comment]
Updated structure to pull out the accessor for the testing purposes

File: lib/pixelsDisplayed/index.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  // At this point, the only remaining case is that the left of the DOM node is
+  // somewhere in the viewport. We can determine the maximum possible amount
+  // displayed rather easily, by determining how much distance there is between
+  // the left of the bounding rectangle and the viewport's width.
+  // However, if this is larger than our DOM node, we don't want to report more
+  // displayed than the node's full width;
+  return Math.min(windowWidth - rect.left, rect.width);
+}
+
+module.exports = { getDimensions: getDimensions,
```
Comment (reason-bv) [inline lib/pixelsDisplayed/index.js]: If you do this then if will never update as the window updates. You should export the functions.

Comment (reason-bv) [inline lib/pixelsDisplayed/index.js]: Ah, never mind, they are the functions.

File: test/unit/pixelsDisplayed/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+
+  var el = {
+    getBoundingClientRect: function () {
+      return rect;
+    }
+  };
+
+  function stubAccessor () {
+    sinonSandbox = sinon.sandbox.create();
+    spy = sinonSandbox.spy();
```
Comment (reason-bv) [inline test/unit/pixelsDisplayed/index.spec.js]: What's the intent here; the stub interface includes the spy interface, so anything you want to do with spy can be done with the stub.

File: test/unit/pixelsDisplayed/index.spec.js [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+      height: 200,
+      top: 100,
+      width: 200,
+      left: 100
+    };
+  });
+
+  afterEach(function () {
+    // we do expect spy to have been called during each test
+    expect(spy.called).to.eql(true);
```
Comment (reason-bv) [inline test/unit/pixelsDisplayed/index.spec.js]:
```
sinon.assert.called(module.getDimensions)
```

File: lib/pixelsDisplayed/README.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+if (pixelsDisplayedVertical(element) > 0) {
+  // element is vertically within viewport
+  if (pixelsDisplayedHorizontal(element) > 0) {
+    // element is both vertically and horizontally within viewport
+  }
+}
+```
+
+## License
```
Comment (reason-bv) [inline lib/pixelsDisplayed/README.md]: The license block can be deleted - the license for the whole package is in the upper directory.

**Comment by reason-bv** [comment]
:+1: Aside from items noted.

**Comment by msmolev** [comment]
Corrected the license notice

**Pull request closed without merging**